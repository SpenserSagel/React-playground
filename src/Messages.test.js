import React from 'react';
import ReactDOM from 'react-dom';
import Messages from './Messages.js';
import renderer from 'react-test-renderer';
import { exportAllDeclaration } from '@babel/types';

describe("Messages", () =>{
    it("RendersWithoutCrashing", () => {
        const div = document.createElement('div');
        ReactDOM.render(<Messages />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it("RendersAsExpected", () => {
        const tree = renderer.create(<Messages name="Messages" unread={4}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("RendersWithNoUnread", () => {
        const tree = renderer.create(<Messages name="Messages" unread={0}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
    });
});