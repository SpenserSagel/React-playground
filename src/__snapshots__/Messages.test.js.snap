// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages RendersAsExpected 1`] = `
<div
  className="messages"
>
  Messages
  <div
    className="unread_count badge"
  >
    4
  </div>
</div>
`;

exports[`Messages RendersWithNoUnread 1`] = `
<div
  className="messages"
>
  Messages
</div>
`;
